lib/mongo/auth/scram.rb

```diff
@@ -49,12 +63,14 @@ def initialize(user)
       #
       # @param [ Mongo::Connection ] connection The connection to log into.
       #   on.
+      # @param [ String ] mechanism The auth mechanism to use (either 'SCRAM-SHA-1' or
+      #   'SCRAM-SHA-256');
       #
       # @return [ Protocol::Message ] The authentication response.
       #
       # @since 2.0.0
-      def login(connection)
-        conversation = Conversation.new(user)
+      def login(connection, mechanism = user.mechanism || :scram)
```

Comment:
It is better to default mechanism to nil here and apply the default in method body. That way downstream code can pass in a mechanism of nil and get the default applied.